spring-kafka/README.md

```diff
@@ -0,0 +1,357 @@
+inject the example bean `kafkaProxyProducerKafkaTemplate` in our applications beans like this:
+```xml
+  <bean id="ourApplicationBean" class="com.hitachivantara.ApplicationBean">
+    <property name="kafkaTemplate" ref="kafkaProxyProducerKafkaTemplate"/> <!-- type converter used on this argument -->
+  </bean>
+``` 
+```java
+public class ApplicationBean {
+  private KafkaTemplate<String, Message> kafkaTemplate;
+  
+  public void setKafkaTemplate( KafkaTemplate<String, Message> kafkaTemplate ) {
+    this.kafkaTemplate = kafkaTemplate;
+  }
+}
+```
+
+### KafkaTemplate
+The easiest way to send messages is using the Spring Kafka Template API defined in `org.springframework.kafka.core.KafkaTemplate`.
+That class has two constructors. Both accept as the first argument an interface `org.apache.kafka.clients.producer.Producer` implementation instance
+and the second optional argument is a `boolean` to configure the auto flush ability of the Kafka Template instance.
+
+#### Producer Factory
+Spring Kafka provides a default implementation for that interface `org.springframework.kafka.core.DefaultKafkaProducerFactory`. We
+default to use that, for which we must provide an `HashMap` with the Kafka connection and library properties.
+
+Those properties are defined in this example in the `kafkaProducerProperties`, and notice that only the basic ones are being used.
+Kafka and Spring Kafka documentation should be used to check which properties exist and how they may influence the application.
+
+#### Producer Key & Value Serializer
+When we send messages through Kafka we can use the Kafka provided serializers for `org.apache.kafka.common.serialization.IntegerSerializer`, 
+`org.apache.kafka.common.serialization.StringSerializer`, etc. But since we advocate the usage of protobuf we must use binary transport
+definition but with specific serializers for Protobuf.
+
+For this end we created a `org.apache.kafka.common.serialization.Serializer<T>` interface implementation in 
+`org.hitachivantara.spring.kafka.protobuf.KafkaProtobufSerializer`. The usage of this Serializer implementation allows us to be able
+to send Protobuf messages directly through the Kafka Template like this:
+```java
+MyProtobufMessage messageBody = MyProtobufMessage
+  .newBuilder()
+  .setValue("Hello World!")
+  .build();
+
+kafkaTemplate.send(
+  "KafkaTopic",
+  "KEY_1",
+  messageBody
+);
+```
+Notice how the `send` method second parameter accepts a String because we used Kafka `StringSerializer` for the key, and the
+third parameter accepts a `com.google.protobuf.MessageLite` because we used our `KafkaProtobufSerializer` for the value.
+
+## Consumer
+
+The consumer part of the messages follows some of the patterns used in the producer but adds a layer of complexity because
+we may have different strategies to read messages from the Kafka brokers. To consume messages from Kafka topics the following 
+blueprint definitions should be used:
```

Comment:
Clearly separate what is part of the concrete example of using `blueprint` from the general usage.

Reply:
It's hard to comply with that request because defining the blueprint and wiring everything together is part of the solution.